bot/extensions/polls/views.py

```diff
@@ -0,0 +1,148 @@
+import discord
+from discord import ui
+
+
+class PollModal(ui.Modal):
+    name = ui.TextInput(label="Choice name", placeholder="Enter poll choice", max_length=32, required=True)
+    description = ui.TextInput(
+        label="Choice description (optional)",
+        placeholder="Enter poll choice description",
+        style=discord.TextStyle.long,
+        max_length=512,
+        required=False,
+    )
+
+    def __init__(self, var: discord.Interaction):
+        self.var = var
+        super().__init__(title="Poll options")
+
+    @property
+    def reactions(self):
+        return {
+            0: "1️⃣",
+            1: "2️⃣",
+            2: "3️⃣",
+            3: "4️⃣",
+            4: "5️⃣",
+            5: "6️⃣",
+            6: "7️⃣",
+            7: "8️⃣",
+            8: "9️⃣",
+            9: "🔟",
+        }
+
+    async def on_submit(self, interaction: discord.Interaction) -> None:
+        message = await self.var.followup.fetch_message(self.var.message.id)
+
+        num = len(message.embeds[0].fields)
+        message.embeds[0].add_field(
+            name=f"{str(self.reactions[num])}  {self.name}", value=self.description, inline=False
+        )
+
+        await message.edit(embed=message.embeds[0])
+        await interaction.response.defer()
+
+
+class CreatePollView(ui.View):
+    CREATE_CUSTOM_ID = "extensions:polls:create"
+    ADD_CUSTOM_ID = "extensions:polls:add"
+    SELECT_CUSTOM_ID = "extensions:polls:select"
+    DELETE_CUSTOM_ID = "extensions:polls:delete"
+
+    def __init__(self, *, timeout=180):
+        super().__init__(timeout=timeout)
+
+    @property
+    def reactions(self):
+        return {
+            0: "1️⃣",
+            1: "2️⃣",
+            2: "3️⃣",
+            3: "4️⃣",
+            4: "5️⃣",
+            5: "6️⃣",
+            6: "7️⃣",
+            7: "8️⃣",
+            8: "9️⃣",
+            9: "🔟",
+        }
+
+    @discord.ui.button(label="Add Choice", style=discord.ButtonStyle.gray, emoji="➕", custom_id=ADD_CUSTOM_ID)
```

Comment:
Weird indent?

Reply:
im not quite sure what you mean? Can you clarify